### PR TITLE
Improve dependencies management

### PR DESCRIPTION
**Pull request by yuri-sergiichuk**
This PR improves our Gradle dependencies management a bit and applies the new scope to the dependencies that are used by the code analysis software only.

As part of the PR, I have also eliminated the usage of the `hamcrest` assertion framework in favor of `truth`.

The change is related to the https://github.com/SpineEventEngine/config/issues/156 issue.

### CODE REVIEW

File: datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java [hunk 1]
```diff
@@ -45,14 +43,14 @@ void testProduceBCBuilder() {
         DatastoreStorageFactory factory = givenFactory();
         BoundedContextBuilder builder = BoundedContext.multitenant(
                 NewBoundedContextBuilderTest.class.getName());
-        assertFalse(builder.tenantIndex()
-                           .isPresent());
+        assertThat(builder.tenantIndex()).isEmpty();
```
Comment (dmdashenkov) [inline datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java]:
```suggestion
        assertThat(builder.tenantIndex())
                .isEmpty();
```

File: datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java [hunk 2]
```diff
@@ -45,14 +43,14 @@ void testProduceBCBuilder() {
...
 
         Optional<? extends TenantIndex> updatedIndex =
                 factory.configureTenantIndex(builder)
                        .tenantIndex();
-        assertTrue(updatedIndex.isPresent());
-        assertThat(updatedIndex.get(), instanceOf(TestNamespaceIndex.getType()));
+        assertThat(updatedIndex).isPresent();
+        Truth.assertThat(updatedIndex.get())
```
Comment (dmdashenkov) [inline datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java]: Please use a static import. Java can handle multiple imported methods with the same name if parameters are different.

File: datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java [hunk 3]
```diff
@@ -45,14 +43,14 @@ void testProduceBCBuilder() {
...
-        assertFalse(builder.tenantIndex()
-                           .isPresent());
+        assertThat(builder.tenantIndex()).isEmpty();
 
         Optional<? extends TenantIndex> updatedIndex =
                 factory.configureTenantIndex(builder)
                        .tenantIndex();
-        assertTrue(updatedIndex.isPresent());
-        assertThat(updatedIndex.get(), instanceOf(TestNamespaceIndex.getType()));
+        assertThat(updatedIndex).isPresent();
```
Comment (dmdashenkov) [inline datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java]: Please apply the formatting as in the suggestion above — the assertion on a new line. It's our conversion. It makes assertions easier to read.

File: datastore/src/test/java/io/spine/server/storage/datastore/tenant/SingleTenantNamespaceSupplierTest.java [hunk 4]
```diff
@@ -56,40 +50,24 @@ void testProduceCustom() {
         String namespaceValue = "my-custom-namespace";
         NamespaceSupplier supplier = NamespaceSupplier.singleTenant(namespaceValue);
         Namespace namespace = supplier.get();
-        assertNotNull(namespace);
-        assertEquals(namespaceValue, namespace.value());
+        assertThat(namespace).isNotNull();
+        assertThat(namespace.value()).isEqualTo(namespaceValue);
```
Comment (dmdashenkov) [inline datastore/src/test/java/io/spine/server/storage/datastore/tenant/SingleTenantNamespaceSupplierTest.java]: Same in this file.

**Review by dmdashenkov — APPROVED** [review]
@yuri-sergiichuk, LGTM with a few minor comments.

File: datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java (hunk 2 above)
Comment (yuri-sergiichuk) [reply datastore/src/test/java/io/spine/server/storage/datastore/NewBoundedContextBuilderTest.java]: What's interesting is that IDEA does not allow fixing if not doing the static import manually.

**Pull request merged**